From 53422bdebcf8721d11fd45d27e3c8149da786846 Mon Sep 17 00:00:00 2001
From: Dietmar Eggemann <dietmar.eggemann@arm.com>
Date: Mon, 14 Dec 2015 12:45:14 +0000
Subject: [PATCH 58/63] arm64: Make defconfig suitable for schedtest

Make sure defconfig produces a JUNO system suitable for schedtest

Changes in:

* task scheduler/task groups
* ftrace

Created with:

* make defconfig
* make menuconfig ... apply changes
* make savedefconfig
* cp .config arch/arm64/configs/defconfig

Signed-off-by: Dietmar Eggemann <dietmar.eggemann@arm.com>
---
 arch/arm64/configs/defconfig | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)
 mode change 100755 => 100644 arch/arm64/configs/defconfig

diff --git a/arch/arm64/configs/defconfig b/arch/arm64/configs/defconfig
old mode 100755
new mode 100644
index e4623a2..dcbb51d
--- a/arch/arm64/configs/defconfig
+++ b/arch/arm64/configs/defconfig
@@ -14,14 +14,15 @@ CONFIG_TASK_IO_ACCOUNTING=y
 CONFIG_IKCONFIG=y
 CONFIG_IKCONFIG_PROC=y
 CONFIG_LOG_BUF_SHIFT=14
+CONFIG_CGROUPS=y
 CONFIG_MEMCG=y
 CONFIG_MEMCG_SWAP=y
 CONFIG_MEMCG_KMEM=y
 CONFIG_CGROUP_HUGETLB=y
+CONFIG_CGROUP_SCHED=y
 # CONFIG_UTS_NS is not set
 # CONFIG_IPC_NS is not set
 # CONFIG_NET_NS is not set
-CONFIG_SCHED_AUTOGROUP=y
 CONFIG_BLK_DEV_INITRD=y
 CONFIG_KALLSYMS_ALL=y
 # CONFIG_COMPAT_BRK is not set
@@ -242,12 +243,19 @@ CONFIG_KVM=y
 CONFIG_PRINTK_TIME=y
 CONFIG_DEBUG_INFO=y
 CONFIG_DEBUG_FS=y
+CONFIG_HEADERS_CHECK=y
+CONFIG_DEBUG_SECTION_MISMATCH=y
 CONFIG_MAGIC_SYSRQ=y
 CONFIG_DEBUG_KERNEL=y
 CONFIG_LOCKUP_DETECTOR=y
 # CONFIG_SCHED_DEBUG is not set
+CONFIG_SCHEDSTATS=y
 # CONFIG_DEBUG_PREEMPT is not set
-# CONFIG_FTRACE is not set
+CONFIG_FTRACE=y
+CONFIG_FUNCTION_TRACER=y
+CONFIG_IRQSOFF_TRACER=y
+CONFIG_PREEMPT_TRACER=y
+CONFIG_SCHED_TRACER=y
 CONFIG_MEMTEST=y
 CONFIG_SECURITY=y
 CONFIG_CRYPTO_ANSI_CPRNG=y
-- 
1.9.1

